src/Farfetch.LoadShedding/Tasks/ConcurrentCounter.cs

```diff
@@ -64,10 +64,7 @@ public int Decrement()
         {
             lock (this._locker)
             {
-                if (this._count > 0)
-                {
-                    this._count--;
-                }
+                this._count--;
```

Comment:
Why remove the if condition? What will happen if the count comes as 0?

Reply:
In the stable version, the counter may be less than zero when an item is rejected because the decrement was done twice for the same item. It was one of the fixes. If there is a risk of the counter being less than zero, we should investigate to fix the root cause. what do you think?

Reply:
Looks good. Thank you!